src/cli/index.ts

```diff
@@ -59,19 +59,24 @@ async function mainWorker(d: D) {
 
     const start = performance.now();
 
-    let errored = false;
+    const runner = new Runner(d);
     try {
-        const runner = new Runner(d);
         await runner.runTasks(...tasks);
     } catch {
-        errored = true;
         // We will have already printed some message here.
         // Set the error code and let the process run to completion,
         // so we don't end up with an unflushed output.
         d.setExitCode(1);
     } finally {
         const took = performance.now() - start;
-        d.log(`Completed ${taskNames}${errored ? pc.red(" with errors") : ""} in ${d.prettyMilliseconds(took)}`);
+        d.log(
+            `Completed ${taskNames}${runner.failedTasks.length > 0 ? pc.red(" with errors") : ""} in ${
+                d.prettyMilliseconds(took)
+            }`,
+        );
+        for (const failedTask of runner.failedTasks) {
+            d.log(` - ${pc.red(failedTask.name)} ${failedTask.error}\n`);
```

Comment:
I think my biggest problem with this is that the errors can be arbitrarily large. For example, in `typescript-go`, I use `execa` to run stuff, but its thrown error is a big message with output, exit codes, etc, and I suspect will get very long / misindented here.

If you can, it would be helpful to try throwing a multi-line error in a test and see what happens; maybe it's fine?

Reply:
I'm not committed to any format, I'd be happy with just printing the task names, like `runner.failedTasks.join(", ")`

Reply:
Printing out a list of failed tasks by name (sorted?) seems totally reasonable as a start, yeah.

Reply:
Sorted alpha would be easy, sorted topographically by task would be more complicated.

Reply:
@everett1992 having written a similar task-runner myself, i'd suggest sorting alphabetically. as soon as you have more than a handful of tasks, topologically sorted is almost as confusing as random order :)